src/ol/control/Attribution.js

```diff
@@ -192,10 +200,16 @@ class Attribution extends Control {
    */
   collectSourceAttributions_(frameState) {
     const layers = this.getMap().getAllLayers();
-    const visibleAttributions = Array.from(
+    let visibleAttributions = Array.from(
       new Set(layers.flatMap((layer) => layer.getAttributions(frameState))),
     );
 
+    if (this.staticAttribution_ !== undefined) {
+      visibleAttributions = Array.from(
+        new Set([...visibleAttributions, this.staticAttribution_]),
+      );
+    }
```

Comment:
Just skip the Array.from all and store the Set?
```suggestion
    const visibleAttributions = new Set(layers.flatMap((layer) => layer.getAttributions(frameState)));
    if (this.staticAttribution_ !== undefined) {
      visibleAttributions.add(this.staticAttribution_);
    }
```

and at the end:
```js
    return Array.from(visibleAttributions);
```

Reply:
Since I've now changed it to handle an array of strings now, this has been refactored, taking your advice.